sklearn/preprocessing/_encoders.py

```diff
@@ -722,14 +741,26 @@ def get_feature_names_out(self, input_features=None):
         cats = self.categories_
         input_features = _check_feature_names_in(self, input_features)
 
+        name_combiner = self._get_feature_name_combiner()
         feature_names = []
         for i in range(len(cats)):
-            names = [input_features[i] + "_" + str(t) for t in cats[i]]
+            names = [name_combiner(input_features[i], t) for t in cats[i]]
             if self.drop_idx_ is not None and self.drop_idx_[i] is not None:
                 names.pop(self.drop_idx_[i])
             feature_names.extend(names)
         return np.asarray(feature_names, dtype=object)
 
+    def _get_feature_name_combiner(self):
+        if self.feature_name_combiner == "concat_string":
+            return lambda feature, category: feature + "_" + str(category)
+        elif callable(self.feature_name_combiner):
```

Comment:
Should we also test that the callable returns a string?

Reply:
Do you mean testing during runtime, and raising if that's not the case?

Reply:
I meant we could test here inside `_get_feature_name_combiner` that a simple input like `self.feature_name_combiner("feature", "category")` produces a string.

@thomasjpfan What do you think? Do we need such a check or is it over the top?

Reply:
I don't know the sklearn direction regarding types, but if it's planning to introduce types at some point, the type system should be the one to check such things.

Of course, users will be ones who will chose to opt in or out of checks.

Reply:
We will not have a type system right now. So it would be safer to test